federationsender/queue/queue.go

```diff
@@ -128,14 +131,33 @@ func (oqs *OutgoingQueues) SendEvent(
 		)
 	}
 
-	// Remove our own server from the list of destinations.
-	destinations = filterAndDedupeDests(oqs.origin, destinations)
-	if len(destinations) == 0 {
+	// Deduplicate destinations and remove the origin from the list of
+	// destinations just to be sure.
+	destmap := map[gomatrixserverlib.ServerName]struct{}{}
```

Comment:
`map[gomatrixserverlib.ServerName]bool` would be slightly better? Reads nicer too.

Reply:
I mostly picked an empty struct because they don't occupy memory.